### PR TITLE
Lock TCP socket when writing

### PR DESCRIPTION
**Pull request by shashi**
without this, many tasks writing at the same time was causing chaos.

### CODE REVIEW

**Comment by shashi** [comment]
@tkelman any idea what is up with the travis test against Julia nightly? https://travis-ci.org/JuliaWeb/WebSockets.jl/jobs/83081375 `ERROR: GitError(Code:ECERTIFICATE, Class:SSL, The SSL certificate is invalid)`

File: src/WebSockets.jl [hunk 1]
```diff
@@ -94,22 +96,27 @@ function write_fragment(io::IO, islast::Bool, data::Array{UInt8}, opcode)
...
+      write(io, data)
+    elseif l <= typemax(UInt64)
+      write(io, b1)
+      write(io, @compat UInt8(127))
+      write(io, hton(@compat UInt64(l)))
+      write(io, data)
+    else
+      error("Attempted to send too much data for one websocket fragment\n")
+    end
+  finally
```
Comment (amitmurthy) [inline src/WebSockets.jl]: You can optimize the multiple writes above by calling `Base.buffer_writes` once on the socket at construction time (e.g. https://github.com/JuliaLang/julia/blob/3f8d5a9aabfc76e445ce5d58875b4327c525454e/base/multi.jl#L145) and then finally flushing once after all individual writes of a fragment - e.g. https://github.com/JuliaLang/julia/blob/3f8d5a9aabfc76e445ce5d58875b4327c525454e/base/multi.jl#L227

This results in the TCPSocket aggregating individual writes into a local buffer (upto 65K) and actually writing to the socket upon a `flush`. Larger than 65K writes get written immediately.

Comment (shashi) [inline src/WebSockets.jl]: Do you know of an alternative to `buffer_writes` on Julia 0.3?

Comment (amitmurthy) [inline src/WebSockets.jl]: No, but then 0.3 does not have `lock` either. For 0.3, you can do multiple writes to a IOBuffer, followed by a takebuf_array and write the array once to the socket. The IOBuffer writes will not yield to other tasks, so you will not need a lock.

Comment (shashi) [inline src/WebSockets.jl]: I did this, thanks!!

**Pull request merged**